### ISSUE TITLE
[charts] Support Waterfall

### ISSUE THREAD

**Issue by oliviertassinari** [mui/mui-x#11318]
### Summary 💡

An issue to test the interest for this feature.

<img src="https://github.com/mui/mui-x/assets/3165635/7bf6a17e-78e2-450d-a062-5c88fe388ac2" width="787">

### Examples 🌈

- https://docs.anychart.com/Basic_Charts/Waterfall_Chart
- https://echarts.apache.org/handbook/en/how-to/chart-types/bar/waterfall/
- https://demos.telerik.com/kendo-ui/waterfall-charts/index
- https://charts.ag-grid.com/react/waterfall-series/
- https://www.highcharts.com/docs/chart-and-series-types/waterfall-series
- Real-life use case https://alexandre.substack.com/p/unpacking-instacarts-ipo

<img src="https://github.com/mui/mui-x/assets/3165635/66c3337e-76a5-4ccb-aa4c-ae088b0b8b05" width="694">

**Search keywords**: Waterfall

**Comment by prakhargupta1** [comment]
Isn't this somewhat already supported in [this](https://mui.com/x/react-charts/bars/#color-scale) demo when you choose both color maps as piecewise.

**Comment by bernardobelchior** [comment]
I don't think so. In that demo, all bars start at 0. In a waterfall chart, the bar starts at the end of the previous one.

**Comment by bernardobelchior** [comment]
While we don't yet have a `WaterfallChart` component, this can be implemented using a Range Bar Chart. We even have a demo [here](https://mui.com/x/react-charts/bar-demo/#waterfall-chart) on how to do it.
